### PR TITLE
KEYCLOAK-516 : Automatic creation of RH-SSO quickstart branch

### PR DESCRIPTION
**Pull request by sebastienblanc**
For now, It will be pushed to https://github.com/redhat-developer/redhat-sso-quickstarts/blob/production-test , we can later decide where the final branch will be.
Travis has been configured to perform the git and push tasks. The producization script should not run when it's a PR (so only on Push to master).

### CODE REVIEW

**Comment by abstractj** [comment]
@sebastienblanc I'm confused, should we merge it on master? Or `product_script`? Plus, we must check what's wrong on Travis.

**Comment by sebastienblanc** [comment]
Yes it will be merged to master really sonnish. Travis is complaining about Angular2 QuickStart and is (I think) unrelated to this PR.

Envoyé de mon iPhone

**Comment by abstractj** [comment]
@sebastienblanc yep, it is unrelated. But before any merge, we need to fix this.

**Comment by sebastienblanc** [comment]
@ssilvert Hi Stan, could you check the travis logs ? Maybe you have an idea on what is going on with the Angular2 quickstart.

**Comment by abstractj** [comment]
@sebastienblanc I don't think it's related with Angular.js. See my changes here https://github.com/keycloak/keycloak-quickstarts/pull/53#partial-pull-merging. Even skipping Angular.js, we still the build fails.

File: travis-run-tests.sh [hunk 1]
```diff
@@ -25,5 +25,10 @@ if [ $1 == "group4" ]; then
   cd ../app-springboot
   mvn -B -s ../maven-settings.xml clean test
 fi
```
Comment (abstractj) [inline travis-run-tests.sh]: @sebastienblanc out of curiosity. Why you have 2 ifs for `group5`?

Comment (sebastienblanc) [reply travis-run-tests.sh]: `productize.sh` should not run for Pull Requests.

Comment (abstractj) [reply travis-run-tests.sh]: @sebastienblanc why not something like:
```
if [ $1 == "group5" ] && [ $TRAVIS_PULL_REQUEST == "false" ]; then
 ./productize.sh
  exit 0
fi  
```

???

**Comment by abstractj** [comment]
@sebastienblanc fyi after take Angular.js app out of the equation. I get another error on travis with your PR.

- Logs: https://api.travis-ci.org/jobs/289901152/log.txt?deansi=true
- Job: https://travis-ci.org/abstractj/keycloak-quickstarts/jobs/289901152
- Changes at your branch: https://github.com/abstractj/keycloak-quickstarts/commits/product_script

Error message:

```
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <{GH_USER_EMAIL}>) not allowed
fatal: unable to access 'https://@/': Could not resolve host: 

travis_time:end:032e99b9:start=1508410690039481645,finish=1508410728776867224,duration=38737385579
[0K
[31;1mThe command "./travis-run-tests.sh $TESTS" exited with 128.[0m
travis_fold:start:before_cache
[0Ktravis_time:start:0cdfb1e9
[0K$ rm -rf $HOME/.m2/repository/org/keycloak/quickstarts
```

Feel free to apply my changes [here](https://github.com/keycloak/keycloak-quickstarts/pull/54) to see the error or exclude Angular.js example for testing.

File: travis-run-tests.sh (hunk 1 above)
Comment (sebastienblanc) [reply travis-run-tests.sh]: yeah, probably to my incremental experiences, the unecessary condition has not been removed, let me try your suggestion.

Comment (sebastienblanc) [reply travis-run-tests.sh]: Yeah let me try that, the extra condition was probably there after my (many) expirements.

**Comment by sebastienblanc** [comment]
@abstractj It will probably not work on your travis account, several en variables are needed for this script to run : 
envvar

**Comment by abstractj** [comment]
@sebastienblanc try to rebase against the latest changes on master. If the PR goes green, merge it and call it a day.

**Comment by sebastienblanc** [comment]
Ok, travis is happy again, Should I squash these commits ? And then I think it's good to merge.

**Comment by abstractj** [comment]
@sebastienblanc yes, and go ahead.

**Pull request merged**